src/pointer.rs

```diff
@@ -738,16 +828,114 @@ impl core::fmt::Display for PointerBuf {
+            tok_offset = 0;
+            ptr_offset = offset;
+            continue;
+        }
+        tok_offset += 1;
+        // if the character is a '~', then the next character must be '0' or '1'
+        // otherwise the encoding is invalid and `InvalidEncodingError` is returned
+        if c == b'~' {
+            // pulling down
+            let next = bytes.next().map(|(_, c)| c);
+            if !matches!(next, Some(b'0' | b'1')) {
+                // the pointer is not properly encoded
+                //
+                // we use the pointer offset, which points to the last
+                // encountered seperator, as the offset of the error.
+                // The source `InvalidEncodingError` then uses the token
+                // offset.
+                //
+                // "/foo/invalid~encoding"
+                //      ^       ^
+                //      |       |
+                //  ptr_offset  |
+                //          tok_offset
+                //
+                return Err(ParseError::InvalidEncoding {
+                    offset: ptr_offset,
+                    source: InvalidEncodingError { offset: tok_offset },
+                });
+            }
+            tok_offset += 1;
+        }
+    }
+    Ok(value)
+}
+
+unsafe fn extend_one_before(s: &str) -> &str {
+    let ptr = s.as_ptr().offset(-1);
+    let len = s.len() + 1;
+    let slice = slice::from_raw_parts(ptr, len);
+    core::str::from_utf8_unchecked(slice)
+}
+
+const fn is_valid_ptr(value: &str) -> bool {
+    let bytes = value.as_bytes();
+
+    if bytes.is_empty() {
+        // root pointer
+        return true;
+    }
+
+    match bytes[0] {
+        b'/' => {}
+        _ => return false,
+    }
+
+    let mut i = 0;
+    while i < bytes.len() {
+        if bytes[i] == b'~' {
+            if i + 1 >= bytes.len() {
+                return false;
+            }
+            if bytes[i + 1] != b'0' && bytes[i + 1] != b'1' {
+                return false;
+            }
+        }
+        i += 1;
+    }
+
+    true
+}
+
+/*
+░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+╔══════════════════════════════════════════════════════════════════════════════╗
+║                                                                              ║
+║                                    Tests                                     ║
+║                                   ¯¯¯¯¯¯¯                                    ║
+╚══════════════════════════════════════════════════════════════════════════════╝
+░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+*/
+
 #[cfg(test)]
 mod tests {
     use super::*;
     use quickcheck::TestResult;
     use quickcheck_macros::quickcheck;
 
     #[test]
-    #[should_panic]
+    #[should_panic = "invalid JSON Pointer"]
     fn from_const_validates() {
-        Pointer::from_static("foo/bar");
+        let _ = Pointer::from_static("foo/bar");
```

Comment:
I know I told you we should prefer `from_static`, but `must_parse` reads a bit better when used as a validator... And it makes sense to not make it `must_use`. Consider this a nit though.

Reply:
Hah, I was just about to rip out `must_parse`.

Hmm.

Reply:
Well arguably we could also just use `Pointer::parse(...).unwrap()`, which is also pretty idiomatic if a little more verbose. Up to you.

Reply:
I took out `must_parse`, left `from_static`.